nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
-                splits.add(split);
+            if (error.get()) {
+                session.transfer(flowFile, REL_FAILURE);
+            } else {
+                session.transfer(flowFile, REL_ORIGINAL);
+                session.transfer(splitFlowFiles, REL_SPLITS);
             }
-        }
-        finishFragmentAttributes(session, flowFile, splits);
-
-        if (splits.size() > 10) {
-            logger.info("Split {} into {} files", new Object[]{flowFile, splits.size()});
         } else {
-            logger.info("Split {} into {} files: {}", new Object[]{flowFile, splits.size(), splits});
+            context.yield();
         }
-
-        session.transfer(flowFile, REL_ORIGINAL);
-        session.transfer(splits, REL_SPLITS);
     }
 
-    private void finishFragmentAttributes(final ProcessSession session, final FlowFile source, final List<FlowFile> splits) {
-        final String originalFilename = source.getAttribute(CoreAttributes.FILENAME.key());
-
-        final String fragmentId = UUID.randomUUID().toString();
-        final ArrayList<FlowFile> newList = new ArrayList<>(splits);
-        splits.clear();
-        for (int i = 1; i <= newList.size(); i++) {
-            FlowFile ff = newList.get(i - 1);
-            final Map<String, String> attributes = new HashMap<>();
-            attributes.put(FRAGMENT_ID, fragmentId);
-            attributes.put(FRAGMENT_INDEX, String.valueOf(i));
-            attributes.put(FRAGMENT_COUNT, String.valueOf(newList.size()));
-            attributes.put(SEGMENT_ORIGINAL_FILENAME, originalFilename);
-            FlowFile newFF = session.putAllAttributes(ff, attributes);
-            splits.add(newFF);
-        }
+    /**
+     *
+     */
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext validationContext) {
+        List<ValidationResult> results = new ArrayList<>();
+        boolean invalidState = (validationContext.getProperty(LINE_SPLIT_COUNT).asInteger() == 0
+                && !validationContext.getProperty(FRAGMENT_MAX_SIZE).isSet());
+        results.add(new ValidationResult.Builder().subject("Maximum Fragment Size").valid(!invalidState)
+                .explanation("Property must be specified when Line Split Count is 0").build());
+        return results;
     }
 
-    private static class SplitInfo {
-
-        public long offsetBytes;
-        public long lengthBytes;
-        public long lengthLines;
-        public long bufferedBytes;
-        public boolean endOfStream;
-
-        public SplitInfo() {
-            this.offsetBytes = 0L;
-            this.lengthBytes = 0L;
-            this.lengthLines = 0L;
-            this.bufferedBytes = 0L;
-            this.endOfStream = false;
-        }
+    /**
+     *
+     */
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Collections.unmodifiableList(properties);
```

Comment:
Since we already have a List<PropertyDescriptor> as a member variable, why not just make the member variable an unmodifiable list and return that instead of creating a new one each time?